File: Models/Location.cs
namespace RouteMK.Models;

public class Location
{
    public int LocationId { get; set; }

    public decimal Latitude { get; set; }

    public decimal Longitude { get; set; }

    public string Name { get; set; } = null!;

    public virtual ICollection<Route> RouteFromLocations { get; set; } = new List<Route>();

    public virtual ICollection<Route> RouteToLocations { get; set; } = new List<Route>();

    public virtual ICollection<Ticket> TicketGetsOffLocations { get; set; } = new List<Ticket>();

    public virtual ICollection<Ticket> TicketGetsOnLocations { get; set; } = new List<Ticket>();

    public virtual ICollection<TripStop> TripStops { get; set; } = new List<TripStop>();
}

File: src/main/java/mk/route/routemk/models/Location.java
package mk.route.routemk.models;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "location")
public class Location {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "location_id")
    private Integer id;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    @Column(name = "name")
    private String name;

    public Location(Double latitude, Double longitude, String name) {
        this.latitude = latitude;
        this.longitude = longitude;
        this.name = name;
    }

    public Location() {
    }

    public String getName() {
        return name;
    }

    public Double getLongitude() {
        return longitude;
    }

    public Double getLatitude() {
        return latitude;
    }

    public Integer getId() {
        return id;
    }
}
